feature/guarantor/src/commonMain/kotlin/org/mifos/mobile/feature/guarantor/screens/guarantorDetails/GuarantorDetailScreen.kt

```diff
@@ -0,0 +1,129 @@
+import org.mifos.mobile.core.ui.component.MifosProgressIndicatorOverlay
+import org.mifos.mobile.core.ui.utils.EventsEffect
+
+@Composable
+internal fun GuarantorDetailScreen(
+    navigateBack: () -> Unit,
+    updateGuarantor: (index: Int, loanId: Long) -> Unit,
+    modifier: Modifier = Modifier,
+    viewModel: GuarantorDetailViewModel = koinViewModel(),
+) {
+    val state by viewModel.stateFlow.collectAsStateWithLifecycle()
+    val scope = rememberCoroutineScope()
+    val snackbarHostState = remember { SnackbarHostState() }
+
+    EventsEffect(viewModel.eventFlow) { event ->
+        when (event) {
+            GuarantorDetailEvent.NavigateBack -> navigateBack()
+            is GuarantorDetailEvent.ShowToast -> {
+                scope.launch {
+                    snackbarHostState.showSnackbar(event.message)
+                }
+            }
+
+            is GuarantorDetailEvent.UpdateGuarantor -> {
+                updateGuarantor.invoke(event.index, event.loanId)
+            }
+        }
+    }
+
+    GuarantorDetailScreen(
+        state = state,
+        modifier = modifier,
+        onAction = remember(viewModel) {
+            { viewModel.trySendAction(it) }
+        },
+        snackbarHostState = snackbarHostState,
+    )
+}
+
+@Composable
+private fun GuarantorDetailScreen(
+    state: GuarantorDetailState,
+    onAction: (GuarantorDetailAction) -> Unit,
+    snackbarHostState: SnackbarHostState,
+    modifier: Modifier = Modifier,
+) {
+    MifosScaffold(
+        topBar = {
+            GuarantorDetailTopBar(
+                navigateBack = { onAction(GuarantorDetailAction.NavigateBack) },
+                deleteGuarantor = { onAction(GuarantorDetailAction.UpdateDialogValue) },
+                updateGuarantor = { onAction(GuarantorDetailAction.UpdateGuarantor) },
+            )
+        },
+        snackbarHostState = snackbarHostState,
+        content = {
+            Box(modifier = Modifier.padding(it)) {
+                state.guarantor?.let { it1 -> GuarantorDetailContent(data = it1) }
+            }
+
+            if (state.showDialog) {
+                MifosAlertDialog(
+                    onDismissRequest = { onAction.invoke(GuarantorDetailAction.UpdateDialogValue) },
+                    dismissText = stringResource(Res.string.dismiss),
+                    confirmationText = stringResource(Res.string.yes),
+                    dialogTitle = stringResource(Res.string.delete_guarantor),
+                    onConfirmation = {
+                        onAction.invoke(GuarantorDetailAction.DeleteGuarantor)
+                        onAction.invoke(GuarantorDetailAction.UpdateDialogValue)
+                    },
+                    dialogText = stringResource(
+                        Res.string.dialog_are_you_sure_that_you_want_to_string,
+                        stringResource(Res.string.delete_guarantor),
+                    ),
+                )
+            }
+        },
+        modifier = modifier,
+    )
```

Comment:
For good practice move what ever is inside the content{} block to after `modifier = modifier`. It is more readable.
sth like this:
```
....
modifier  = modifier,
) {
 your content code here
}
```